### PR TITLE
feat: use uv instead of poetry

### PR DESCRIPTION
**Pull request by amannocci**
## What is the change being made?

- Use uv instead of poetry for dependencies management.

## Why is the change being made?

- uv also handle python interpreter installation.

### CODE REVIEW

**Review by reakaleek — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,17 +5,6 @@
 
 version: 2
 updates:
-  - package-ecosystem: "pip"
```
Comment (v1v) [inline .github/dependabot.yml:8]: I'm not familiar with the dependency management for `uv`, but how does it work for `uv`? I cannot see any references about keeping the dependencies up to date.

**Review by v1v — COMMENTED** [review]
I'm not an expert of python, but I see the bump dependency being removed, if that's the consequence for using `uv`, I cannot see much value though

**Review by v1v — CHANGES_REQUESTED** [review]
as long as no dependency management in place using Dependabot, I don't think we can use `uv`

**Comment by amannocci** [comment]
> as long as no dependency management in place using Dependabot, I don't think we can use `uv`

At the moment, dependabot doesn't support bumping the uv.lock file.
However, there are many benefits using uv instead of poetry.
Bumping dependencies should not block us to use another dependency management tool.
We can temporary create a small github workflow to issue `uv lock` and create PR with the result.

**Comment by v1v** [comment]
> However, there are many benefits using uv instead of poetry.

I understand it moves the logic from the GH actions to the build system itself. However, can you help me understand what the other benefits are? 🙏 

There is one concern about the uv project, as far as I see, the first release for https://github.com/astral-sh/uv was about 10 months ago in Feb 2024, and they are still in `0.x.x`. While for poetry (https://github.com/python-poetry/poetry), I see they are in version `1.x`, and the first release was in 2018.


> We can temporary create a small github workflow to issue uv lock and create PR with the result.

I'm not a big fan of redoing something from scratch because we migrated one tool for another and removed that feature

**Review by v1v — APPROVED** [review]
We have agreed to support the  uv dependency management with `updatecli` initially

**Pull request merged**